packages/library-legacy/src/connection.ts

```diff
@@ -1178,7 +1208,7 @@ export type ParsedInstruction = {
   /** ID of the program for this instruction */
   programId: PublicKey;
   /** Parsed instruction info */
-  parsed: any;
+  parsed?: any;
```

Comment:
I suppose this change is here for correctness – ie. this property can, in fact, be missing?

I don't think this will make a difference to consumers since this was `any` to begin with ([playground link](https://www.typescriptlang.org/play/?#code/C4TwDgpgBAYg9nKBeKBvAUFLUwEMBOAzhACYBcUuAdiOgL7qiSwIByEA7smptnkaQD8FarQboAxnCqFgUAGYJuVAK4AbNZUJQVVANZU4HKlpZwA3JOmyFbTsvWbc23QaMnnZ9h0vpFcADp+YhIA2RV5eXMoAHoYqAB5AGk-Ow4gghCw4Aio2PiAZWAASw1EpKA)).

Reply:
Actually, is this true? Is `parsed` not always defined, as per this? https://github.com/anza-xyz/agave/blob/5263c9d61f3af060ac995956120bef11c1bbf182/transaction-status/src/parse_instruction.rs#L83

Reply:
https://docs.rs/serde_json/latest/serde_json/enum.Value.html

Reply:
@0xlchigo, any thoughts on this? I really want to slam this in, but this is the last thing I'm stuck on.

Reply:
My apologies for missing your previous messages @steveluscher! The reason why I made it optional was because TypeScript throws some nasty type errors if it isn't set as optional. I agree that I don't think it'll make a difference to consumers — it was more so to make the compiler happy

Reply:
Took a look. It's because `parsed` was wrong in `ParsedInstructionStruct`. It should be `unknown()`.